packages/edit-site/src/components/block-editor/use-site-editor-settings.js

```diff
@@ -92,13 +93,18 @@ function useArchiveLabel( templateSlug ) {
 
 function useGoBack() {
 	const location = useLocation();
+	const previousLocation = usePrevious( location );
 	const history = useHistory();
 	const goBack = useMemo( () => {
 		const isFocusMode =
 			location.params.focusMode ||
-			FOCUSABLE_ENTITIES.includes( location.params.postType );
-		return isFocusMode ? () => history.back() : undefined;
-	}, [ location.params.focusMode, location.params.postType, history ] );
+			( location.params.postId &&
+				FOCUSABLE_ENTITIES.includes( location.params.postType ) );
+		const didComeFromEditorCanvas =
```

Comment:
I wonder if we should make this future proof. It's true that right now the "edit" mode canvas is only used for the editor when editing entities. But nothing stops us from using canvas=edit in other unrelated pages.

So I wonder if we should just for the presence of postId and postType in the previous location to make this more future proof. (It's also true that it's not a 100% guarantee either).

PR LGTM anyway

Reply:
Good idea 👍